pkg/util/quotapool/int_rate_impl_test.go

```diff
@@ -0,0 +1,190 @@
+// Copyright 2020 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package quotapool
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
+)
+
+// Limit defines a rate in terms of quota per second.
+type Limit float64
+
+// RateLimiter implements a token-bucket style rate limiter.
+// It has the added feature that quota acquired from the pool can be returned
+// in the case that they end up not getting used.
+type RateLimiter struct {
+	qp *QuotaPool
+
+	// TODO(ajwerner): synchronization around changing limits.
+	burst     int64
+	rateLimit Limit
+}
+
+// NewRateLimiter defines a new RateLimiter.
+func NewRateLimiter(name string, limit Limit, burst int64, options ...Option) *RateLimiter {
```

Comment:
How does the `burst` behave here? We ran into lots of trouble with stdlib `rate.Limiter` because you weren't allowed to ask more than the burst in any single acquisition (hence https://github.com/cockroachdb/cockroach/blob/76c63ded01f18aa31c08ed6a2e9cf4972b0b0459/pkg/util/limit/rate_limiter.go#L39 came into existence). The mismatch between the underlying types for `rateLimit` and `burst` was also annoying, though it's probably fine if the burst here is better behaved.

Reply:
PS this is also a request to clarify the semantics in the comments (chiefly `NewRateLimiter`), if you don't mind.

Reply:
Also, it seems that we may just remove `*LimiterBurstDisabled` when you're done here, and just use yours.